### PR TITLE
Implemented notifying of followers and reporters via email on issue status change

### PR DESCRIPTION
**Pull request by mis6ko13**
Issue - https://github.com/Kotany-IF070-WebUI-RoR/NewVasjuky/issues/98

- [ ] @Kmalynss  
- [ ] @krisberry
- [x] @igorkasyanchuk 

Added email notifying functionality (on issue status change) for followers and reporters.
Covered with tests.

### CODE REVIEW

File: app/models/follow.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class Follow < Socialization::ActiveRecordStores::Follow
```
Comment (f1nwe) [inline app/models/follow.rb]: What is this? You can find followers as issue.followers

File: app/mailers/issue_mailer.rb [hunk 2]
```diff
@@ -10,4 +8,11 @@ def issue_created(id)
     mail to: emails,
          subject: 'Подано нову скаргу'
   end
+
+  def issue_status_changed(id)
+    @issue = Issue.find(id)
+    @follower_emails = @issue.followers(User).pluck(:email)
+    return unless @follower_emails.any?
+    mail to: @follower_emails.join(';'), subject: 'Змінено статус скарги'
```
Comment (igorkasyanchuk) [inline app/mailers/issue_mailer.rb]: this is not OK ... you will show all user emails in one email in "TO" field.
It's better to send to each user same email + you can say in email "Hello #{@user.first_name}"

File: app/models/follow.rb (hunk 1 above)
Comment (mis6ko13) [inline app/models/follow.rb]: Fixed

File: app/mailers/issue_mailer.rb (hunk 2 above)
Comment (mis6ko13) [inline app/mailers/issue_mailer.rb]: I'll fix that.

File: app/mailers/issue_mailer.rb [hunk 3]
```diff
@@ -10,4 +8,19 @@ def issue_created(id)
...
+    @followers = @issue.followers(User)
+    return unless @followers.any?
+    @followers.each do |follower|
+      mail_to_followers(follower, @issue).deliver
+    end
+  end
+
+  def mail_to_followers(follower, issue)
+    @follower = follower
+    @issue = issue
```
Comment (f1nwe) [inline app/mailers/issue_mailer.rb]: You don't need those variables

File: spec/mailers/issue_mailer_spec.rb [hunk 4]
```diff
@@ -2,14 +2,18 @@
 require 'rails_helper'
 
 describe IssueMailer, type: :mailer do
-  let(:reporter) { create(:user, :reporter) }
-  let!(:issue) { create(:issue) }
-  let(:category) { create(:category) }
-  let(:mail) { IssueMailer.issue_created(issue.id).deliver! }
+  let(:reporter)  { create(:user, :reporter) }
```
Comment (igorkasyanchuk) [inline spec/mailers/issue_mailer_spec.rb]: change `let(:admin)` to `let!(:admin)` and you can remove your  
```
   before :each do
 -    create(:user, :admin)		 +    admin
 -    create(:user, :moderator)		 +    moderator
    end		    end
```

File: app/mailers/issue_mailer.rb (hunk 3 above)
Comment (mis6ko13) [inline app/mailers/issue_mailer.rb]: No I do.
I use them in mail template.

**Pull request merged**